app/src/client/app/BuildPage.tsx

```diff
@@ -1,189 +1,98 @@
-                    'delay-400 absolute left-0 top-2.5 block h-0.5 w-full rounded-sm bg-airt-primary duration-200 ease-in-out dark:bg-white',
-                    {
-                      '!h-0 !delay-200': !sidebarOpen,
-                    }
-                  )}
-                ></span>
-              </span>
-            </span>
-          </button>
-
-          {/* <!-- Hamburger Toggle BTN --> */}
-        </div>
-      </div>
-    </header>
-  );
-};
-
 const BuildPage = ({ user }: BuildPageProps) => {
-  const { data, loading, error } = useBuildPage();
+  const history = useHistory();
+  const { data: propertiesSchema, loading, error } = useBuildPageNew();
   const [sidebarOpen, setSidebarOpen] = useState(false);
-  const [sideNavSelectedItem, setSideNavSelectedItem] = useState('secret');
-  const [togglePropertyList, setTogglePropertyList] = useState(false);
-  const { pathname } = location;
-  const activeBuildPageTab = pathname.split('/').pop();
-
+  const [activeProperty, setActiveProperty] = useState<string | null>(null);
+  const [sideNavItemClickCount, setSideNavItemClickCount] = useState(0);
   const wrapperClass = document.body.classList.contains('server-error')
     ? 'h-[calc(100vh-173px)]'
     : 'h-[calc(100vh-75px)]';
 
-  const history = useHistory();
+  const handleSideNavItemClick = (selectedComponentName: string) => {
+    setActiveProperty(selectedComponentName);
+    setSideNavItemClickCount(sideNavItemClickCount + 1);
+  };
+
+  const setActivePropertyInSessionStorage = (propertyName: string) => {
+    sessionStorage.setItem('activeProperty', propertyName);
+  };
+
   useEffect(() => {
-    if (!user) {
-      history.push('/login');
-    } else {
-      if (!user.hasPaid && user.isSignUpComplete) {
-        history.push('/pricing');
-      }
+    if (propertiesSchema) {
+      const propertyName = sessionStorage.getItem('activeProperty') || propertiesSchema.list_of_schemas[0].name;
+      setActiveProperty(propertyName);
     }
-  }, [user, history]);
+  }, [propertiesSchema]);
 
   useEffect(() => {
-    if (!activeBuildPageTab) return;
-    if (activeBuildPageTab === 'build') {
-      history.push(`/build/secret`);
-    } else {
-      setSideNavSelectedItem(activeBuildPageTab);
+    if (activeProperty) {
+      history.push(`/build/${activeProperty}`);
+      setActivePropertyInSessionStorage(activeProperty);
     }
-  }, [activeBuildPageTab]);
-
-  if (loading) {
-    return <LoadingComponent />;
-  }
+  }, [activeProperty]);
 
-  const handleSideNavItemClick = (selectedComponentName: string) => {
-    setSideNavSelectedItem(selectedComponentName);
-    setTogglePropertyList(!togglePropertyList);
-    history.push(`/build/${selectedComponentName}`);
-  };
+  const canRenderProperty = propertiesSchema && activeProperty;
 
   return (
     <div className='dark:bg-boxdark-2 dark:text-bodydark bg-captn-light-blue'>
+      {loading && <LoadingComponent />}
+      {error && (
+        <p
```

Comment:
Maybe we could consider to factor this out into the separate component.

Reply:
The code related to the error messages are moved to a new file called `components/ErrorComponent.tsx`